genienlp/tasks/almond.py

```diff
@@ -253,15 +254,63 @@ def _make_example(self, parts, dir_name=None, **kwargs):
         # the question is irrelevant
         if len(parts) == 2:
             input_sequence, target_sequence = parts
-            _id = "id-null"
+            example_id = "id-null"
         elif len(parts) == 3:
-            _id, input_sequence, target_sequence = parts
+            example_id, input_sequence, target_sequence = parts
         else:
             raise ValueError(f'Input file contains line with {len(parts)} parts: {str(parts)}')
         question = 'translate from input to output'
         context = input_sequence
         answer = target_sequence
-        return Example.from_raw(self.name + '/' + _id, context, question, answer,
+        return Example.from_raw(self.name + '/' + example_id, context, question, answer,
+                                preprocess=self.preprocess_field, lower=False)
+
+    def get_splits(self, root, **kwargs):
+        return AlmondDataset.return_splits(path=os.path.join(root, 'almond'), make_example=self._make_example, **kwargs)
+
+
+@register_task('paraphrase')
+class Paraphrase(BaseAlmondTask):
```

Comment:
Wouldn't Paraphrase inherit from NaturalSeq2Seq task? Cause future seq2seq tasks like translation will be coming in too and all share similar methods such as metrics, _is_program_field, etc.